### PR TITLE
Add isogram exercise

### PR DESCRIPTION
**Pull request by davespanton**
This adds the isogram exercise example and test files.

### CODE REVIEW

**Comment by catb0t** [comment]
Hey, thanks for the pull request -- I really appreciate it, and it's cool to see people using Factor!

File: exercises/isogram/isogram-example.factor [hunk 1]
```diff
@@ -0,0 +1,5 @@
+USING: kernel ascii sets sequences ;
+IN: isogram
+
+: isogram? ( str -- ? )
+  >lower [ letter? ] filter dup members [ length ] bi@ = ;
```
Comment (catb0t) [inline exercises/isogram/isogram-example.factor]: This pull request on the whole looks excellent, but I do have a cleaner and more Factor-y `isogram?` definition:

```factor
USING: kernel sets ;
: isogram? ( string -- ? )
  dup "" set-like = ;
```

**Comment by catb0t** [comment]
And by the way, now that my school's out, I can finish working on the autogenerator: https://github.com/catb0t/exercism.factor/blob/master/exercism/autogen-exercises/autogen-exercises.factor

It'll alleviate some, but not all, work in the future.

File: exercises/isogram/isogram-example.factor (hunk 1 above)
Comment (davespanton) [inline exercises/isogram/isogram-example.factor]: That's nicer, I've updated with this implementation.

**Comment by catb0t** [comment]
Perfect, thanks! I really appreciate it:D

**Comment by davespanton** [comment]
Glad to help. Look forward to seeing more of your work on the autogenerator.

**Pull request merged**